awebox/ocp/constraints.py

```diff
@@ -174,57 +188,105 @@ def expand_with_collocation(nlp_options, P, V, Xdot, model, Collocation):
 
     mdl_eq_fun = model_constraints_list.get_function(nlp_options, model_variables, model_parameters, 'eq')
     mdl_eq_map = mdl_eq_fun.map('mdl_eq_map', parallellization, n_k * d, [], [])
-    mdl_eq_map_shooting = mdl_eq_fun.map('mdl_eq_map', parallellization, shooting_nodes, [], [])
+
+    mdl_shooting_eq_fun = mdl_shooting_cstr_sublist.get_function(nlp_options, model_variables, model_parameters, 'eq')
+    mdl_shooting_eq_map = mdl_shooting_eq_fun.map('mdl_shooting_eq_map', parallellization, shooting_nodes, [], [])
 
     # evaluate constraint functions
     ocp_ineqs_expr = mdl_ineq_map(shooting_vars, shooting_params)
     ocp_eqs_expr = mdl_eq_map(coll_vars, coll_params)
-    ocp_eqs_shooting_expr = mdl_eq_map_shooting(shooting_vars, shooting_params)
+    ocp_eqs_shooting_expr = mdl_shooting_eq_map(shooting_vars, shooting_params)
 
     # sort constraints to obtain desired sparsity structure
     for kdx in range(n_k):
 
-        # algebraic constraints on shooting nodes
-        cstr_list.append(cstr_op.Constraint(
-            expr = ocp_eqs_shooting_expr[:,kdx],
-            name = 'algebraic_{}'.format(kdx),
-            cstr_type = 'eq'
+        # dynamics on shooting nodes
+        if nlp_options['collocation']['name_constraints']:
+            for cdx in range(ocp_eqs_shooting_expr[:, kdx].shape[0]):
+                cstr_list.append(cstr_op.Constraint(
+                    expr = ocp_eqs_shooting_expr[cdx,kdx],
+                    name = 'shooting_' + str(kdx) + '_' + mdl_shooting_cstr_sublist.get_name_list('eq')[cdx] + '_' + str(cdx),
+                    cstr_type = 'eq'
+                    )
+                )
+        else:
+            cstr_list.append(cstr_op.Constraint(
+                expr=ocp_eqs_shooting_expr[:, kdx],
+                name='shooting_{}'.format(kdx),
+                cstr_type='eq'
+                )
             )
```

Comment:
For me, you can delete the `shooting_{}'.format(kdx)` naming option (and the `'name_constraints'` option altogether), since I do not need this specific naming scheme and all naming schemes are fine for me. Same for path and collocation constraints below.

[edit]: OK I see that it is only used for health debugging :+1: